### PR TITLE
Makes Sous compatible with Go 1.8

### PR DESCRIPTION
**Pull request by samsalisbury**
ext/git: Fix CanonicalRepoURL for go1.8.
- Go 1.8 introduced changes to url.Parse, making it reject
  urls of the format git@github.com:user/project.
- Fix is to prefix these with ssh://.
- See go 1.8 release notes: https://golang.org/doc/go1.8#net_url

### CODE REVIEW

File: ext/git/remote.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ func (rs Remotes) ensureExists(name string) Remote {
 // credentials, and possibly ending with ".git", and returns a clean path of the
 // form: <hostname>/<repo-path>.
 func CanonicalRepoURL(repoURL string) (string, error) {
+	if strings.HasPrefix(repoURL, "git@") {
```
Comment (samsalisbury) [inline ext/git/remote.go]: This assumes GitHub, which uses the 'git' user exclusively. However, it is efficient at getting tests to pass, and makes the same assumptions as other parts of Sous (namely that we are dealing with repos hosted on GitHub). Happy to accept a follow-up PR with a regular expression and more tests to exercise it, but not putting them in this PR, as I just want to make it work so I can move on.

**Comment by samsalisbury** [comment]
Travis? You there?

**Comment by sous-machine-account** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by samsalisbury** [comment]
Closing, this is based on an outdated master :/

**Pull request closed without merging**